### PR TITLE
Update eslint from 8.57.0 to 9.10.0

### PR DESCRIPTION
**Pull request by mosabua**
Replaces #657 

Also adjusts eslint configuration

Bumps [eslint](https://github.com/eslint/eslint) from 8.57.0 to 9.10.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.57.0...v9.10.0)

---
updated-dependencies:
- dependency-name: eslint dependency-type: direct:development update-type: version-update:semver-major ...

Co-autored-by: Manfred Moser <manfred@simpligility.ca>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "version": "0.2.3",
   "name": "trino-client",
   "description": "Trino client library",
+  "type": "module",
```
Comment (nineinchnick) [inline package.json:5]: This should not be required for updating eslint and might have other consequences. It should at least be in a separate commit.

**Review by nineinchnick — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (regadas) [reply package.json:5]: +1

Comment (mosabua) [reply package.json:5]: When it was not there the out said it implied it and had to parse things so it is slower and it should be explicitly declared.

Technically I think it is true that the client is a module right?

Happy to pull it into a separate commit but I think we should still do this.. agreed?

Comment (nineinchnick) [reply package.json:5]: What produces the warning, eslint or node itself? I don't know enough about ES modules to say if it's ok or not, but I don't think it should block the PR either way.

Comment (mosabua) [reply package.json:5]: Move to separate commit now.. ok to merge with it @nineinchnick @regadas ?

Comment (mosabua) [reply package.json:5]: yarn test:lint
(node:35408) [MODULE_TYPELESS_PACKAGE_JSON] Warning: file:///Users/mmoser/dev/github/simpligility/trino-js-client/eslint.config.js?mtime=1727194890625 parsed as an ES module because module syntax was detected; to avoid the performance penalty of syntax detection, add "type": "module" to /Users/mmoser/dev/github/simpligility/trino-js-client/package.json
(Use `node --trace-warnings ...` to show where the warning was created)

/Users/mmoser/dev/github/simpligility/trino-js-client/.yarn/releases/yarn-3.2.1.cjs
  2:1  warning  Unused eslint-disable directive (no problems were reported)

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.

Comment (mosabua) [reply package.json:5]: From that message I conclude that its best to declare explicitly .. since it ends up assuming it anyway.

Comment (mosabua) [reply package.json:5]: I will merge. Worst case we can revert that commit and figure out next steps

Comment (nineinchnick) [reply package.json:5]: When migrating the eslint config, you should have added `--commonjs`:
```bash
npx @eslint/migrate-config .eslintrc.json --commonjs
```

Comment (regadas) [reply package.json:5]: ah I see; sounds good!

**Review by regadas — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (mosabua) [reply package.json:5]: oh .. do we need a follow up change then @nineinchnick ?

Comment (nineinchnick) [reply package.json:5]: We can clean it up to avoid mixing CommonJS and ES module files. I'm not sure what kind of issues it can cause, other than `jest` not being able to work in CommonJS packages that try to import ES module dependencies.

**Comment by yu-iskw** [comment]
@mosabua @regadas I encountered the subsequent issue with v0.2.4 in `@lightdash/cli`. I am assuming that the error is caused by adding `"type": "module"` to `package.json`. I am not familiar with JS/ES/TS very much. Can I have advice to solve this?

https://github.com/lightdash/lightdash/issues/12387

```
$ lightdash
/Users/yu/.anyenv/envs/nodenv/versions/20.10.0/lib/node_modules/@lightdash/cli/node_modules/@lightdash/warehouses/dist/warehouseClients/TrinoWarehouseClient.js:8
const trino_client_1 = require("trino-client");
                       ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/yu/.anyenv/envs/nodenv/versions/20.10.0/lib/node_modules/@lightdash/cli/node_modules/trino-client/dist/index.js from /Users/yu/.anyenv/envs/nodenv/versions/20.10.0/lib/node_modules/@lightdash/cli/node_modules/@lightdash/warehouses/dist/warehouseClients/TrinoWarehouseClient.js not supported.
index.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead either rename index.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /Users/yu/.anyenv/envs/nodenv/versions/20.10.0/lib/node_modules/@lightdash/cli/node_modules/trino-client/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).

    at Object.<anonymous> (/Users/yu/.anyenv/envs/nodenv/versions/20.10.0/lib/node_modules/@lightdash/cli/node_modules/@lightdash/warehouses/dist/warehouseClients/TrinoWarehouseClient.js:8:24) {
  code: 'ERR_REQUIRE_ESM'
}

Node.js v20.10.0
```
